featurebyte/query_graph/sql/specs.py

```diff
@@ -288,6 +289,26 @@ def get_aggregation_source(
     def construct_agg_result_name(self, *args: Any) -> str:
         return super().construct_agg_result_name(*args, self.aggregation_source.query_node_name)
 
+    def get_agg_result_name_from_groupby_parameters(self, parameters: BaseGroupbyParameters) -> str:
+        """
+        Get the name of the aggregation result column from groupby parameters. The name should
+        include the parameters that can affect the result of the groupby operation: aggregation
+        function, parent variable, groupby keys and category key (optional).
+
+        Parameters
+        ----------
+        parameters: BaseGroupbyParameters
+            Groupby parameters
+
+        Returns
+        -------
+        str
+        """
+        args = [parameters.agg_func, parameters.parent]
+        args.extend(parameters.keys or [None])  # type: ignore[list-item]
+        args.append(parameters.value_by or None)
+        return self.construct_agg_result_name(*args)
```

Comment:
This is to handle the case when the same node is used by multiple group by operations (e.g. with different keys).

Reply:
nit: Maybe we could use something like
```
args = [
    parameters.agg_func,
    parameters.parent,
    *(parameters.keys or [None]),
    parameters.value_by or None,
]
```